### ISSUE TITLE
Claims file upload

### ISSUE THREAD

**Issue by srmyers17** [department-of-veterans-affairs/va.gov-team#13205]
## User Story
As a user, I can use my phone's camera to upload files to a claim or add files from my phone's files to a claim.

## Tasks
- [x] High fidelity prototype of screens in iOS
- [ ] Usability test as needed, refine as needed
- [ ] (Ongoing) UI for a range of devices we are supporting, Android-specific designs as needed

## Wireframes
[The document upload UX flow can be triggered from this screen](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432345660).

## UX Documentation 
[Step 1](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623517)
Users should only be able to select from file requests made by VA to upload against. Users should not be able to upload files and name them themselves. A list of file requests from VA should be displayed in the first step of the upload flow.

[Step 2](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623518)
We would like to use React Native plugins that offer access to more apps rather than just the photo roll. Some of our prior tech research indicates this may be feasible. Use iOS or Android native action sheet to select where to upload files from. iOS is shown in the prototype.

[Step 3](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623520)
In the upload flow, we would like to use a React Native plugin to do edge detection. We want to use whatever camera functionality is simplest, assuming that's iOS and Android default camera controls. 

[Step 4](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623521)
Include controls for 'Retake' and 'Keep' after each photo is taken.

[Step 5](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623522)
A user should be able to upload multiple pages to the same file request. Some of our prior tech research indicates that combining multiple images into 1 file is something probably better handled by the API / sending the images to the API and have a flag that tells the API to combine them into a PDF. Even if that's the case, we would like to present the user-facing flow as though they are uploading multiple images into a single file at once. In the case of uploading multiple photos, the 'Save' CTA should update to reflect the total images (e.g. "Save (4)") 

[Step 6](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623523)
If we're able to offer uploading multiple images to one file request at the same time, it would be ideal to have a preview of the full file on the verification screen. @ front-end team: I would love to talk more about the feasibility of this, it seems like a nice to have to me and depending on LOE I could see this being post-MVP.

[Step 7](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623524)
Loading screen. 

[Step 8](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432623525)
Users should see a clear success message after uploading along with two CTAs: one to **Upload another file** (returns users back to the first screen in upload flow) and **Done** (returns user to their claim).

Since we're not sure how quickly the API will register the uploaded docs as registered so we'll have an "uploaded" state for items to signify to users their upload was successful. This is shown in the prototype [here](https://adhoc.invisionapp.com/share/YUYRT9BAPSB#/screens/432737008_Claim-Upload-10).

## Other documentation
1. We won't set page number limits for MVP, users can take as many images as they'd like to.
2. Per the upload tool on VA.gov, accepted file types are: **pdf (unlocked), gif, jpeg, jpg, bmp, txt**. 
3. Maximum file size: 25MB (being increased to 50MB)

**Comment by annaswims** [comment]
The file limit is being increased to 50mb https://github.com/department-of-veterans-affairs/va.gov-team/issues/13178

**Comment by bridgethapner** [comment]
Per 9/30 tech call we'll keep a full list of docs a user can upload until that doc is processed by the backend. We don't know how long that will be. 

We need a UX solve for displaying a doc that has been uploaded but has not been processed yet.

**Comment by bridgethapner** [comment]
Updates: 
- I added a solution for when a file has been uploaded but the API is still displaying it as a needed file. 

@bessgreen Do you think it'd be helpful to have another sync before writing the FE ticket or do you have what you need for now?

**Comment by bridgethapner** [comment]
I added some clearer documentation here that speaks to each step in the flow. Moving this ticket to **Design Review** for now since UX work is done for the time being. I can collaborate as needed on Qs that arise about this feature.

**Comment by annaswims** [comment]
Just to add another element of fun, we're going to start letting users upload pasword-protected PDF along with the password.

**Comment by bridgethapner** [comment]
Good to know, thank you @annaswims! 

I am adding a ticket to our backlog for this...it may not be something we do for Nov MVP (to internal VA audience) but can add in before the app store release. cc @srmyers17

**Comment by bridgethapner** [comment]
Feedback from 10/22 sync with Martha + Steve: 
- Clarify 50 MB limit in UI 
- Look into prioritizing docs by most common types
- Consider adding a message along the lines of "If you are uploading more than 10 pages, desktop will be better"
- Think about framing tool as "Quick Upload" for small files only

**Comment by bridgethapner** [comment]
**Changes:** 
- Added 50 MB file limit and accepted file types text to UI of initial upload screen

**Next Steps:**
- @jmarchi will do a spike here to understand any technical control issues (e.g. can we display a thumbnail of the most recent photo in the camera UI?). 
- Once this technical discovery is done, we will meet to revise the exact scope of the claims file upload feature and complete the UI work. 

cc @srmyers17 FYSA here

**Comment by bridgethapner** [comment]
I am closing this ticket & will create a new ticket for any UI updates required after technical discovery.
